src/sentry/web/api.py

```diff
@@ -181,6 +183,10 @@ def _dispatch(self, request, helper, project_id=None, origin=None,
             if not is_valid_origin(origin, project):
                 raise APIForbidden('Invalid origin: %s' % (origin,))
 
+        if project and not is_valid_ip(request.META['REMOTE_ADDR'], project):
```

Comment:
Going to add a test to cover this path. Even though the logic is simple, it'd be bad if we broke it.